src/sidebar/components/Buttons/index.js

```diff
@@ -0,0 +1,187 @@
+import classnames from 'classnames';
+
+import { SvgIcon } from '@hypothesis/frontend-shared';
+
+/**
+ * @typedef ButtonProps
+ * @prop {Object} [children]
+ * @prop {string} [icon] - name of `SvgIcon` to render in the button
+ * @prop {'left'|'right'} [iconPosition] - Icon positioned to left or to
+ *   right of button text
+ * @prop {boolean} [isDisabled]
+ * @prop {boolean} [isExpanded] - Is the element associated with this button
+ *   expanded (set `aria-expanded`)
+ * @prop {boolean} [isPressed] - Is this button currently "active?" (set
+ *   `aria-pressed`)
+ * @prop {() => any} [onClick]
+ * @prop {'small'|'medium'|'large'} [size] - For styling, size variant
+ * @prop {string} [title]
```

Comment:
I think this is clearer
```suggestion
 * @prop {string} [ariaLabel]
```